cloudbridge/cloud/providers/gce/services.py

```diff
@@ -290,48 +312,54 @@ def get(self, image_name):
             # The API will throw an TypeError, if parameter `image` does not
             # match the pattern "[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?".
             print("TypeError: {0}".format(type_error))
-            return None
         except googleapiclient.errors.HttpError as http_error:
             print("googleapiclient.errors.HttpError: {0}".format(http_error))
-            return None
+        for public_image in self._public_images:
+            if public_image.name == image_name:
+                return public_image
+        return None
 
     def find(self, name, limit=None, marker=None):
         """
         Searches for an image by a given list of attributes
         """
+        if self._public_images == None:
```

Comment:
What do you think about always calling self._retrieve_public_images() and have the check for "None" in that function. That keeps the logic in one place of "None" means we haven't tried to fetch, "a list, even if empty" means we have.